sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala

```diff
@@ -238,20 +238,27 @@ case class CachedRDDBuilder(
   }
 
   def isCachedColumnBuffersLoaded: Boolean = {
-    _cachedColumnBuffers != null && isCachedRDDLoaded
+    if (_cachedColumnBuffers != null) {
```

Comment:
Do you need the double-checked locking anymore? I suppose it's a perf optimization?

Reply:
The double-checked locking follows other methods in the class. Yes, it's a perf optimization.